### ISSUE TITLE
BUG 831833

### ISSUE THREAD

**Issue by jbondc** [Microsoft/TypeScript#3465]
What does the bug make reference to?
https://github.com/Microsoft/TypeScript/blob/9a89147587c06ba51181ff2ee5ade69a98b171ea/tests/cases/conformance/types/typeRelationships/subtypesAndSuperTypes/enumIsNotASubtypeOfAnythingButNumber.ts#L123

**Comment by RyanCavanaugh** [comment]
Old internal bug report. Contents reproduced:

---
```ts
enum E { A }
 
interface I19 {
    [x: string]: Object;
    foo: E; // error, failed in 0.9.1.1
}
 
interface I20 {
    [x: string]: {};
    foo: E; // error, worked in 0.9.1.1
}
 
var x: Object;
x = E.A; // error, worked in 0.9.1.1
var y: {};
y = E.A;

 
function f16<T extends E>(x: T) {
   var r13 = true ? E.A : x; // error
   var r13 = true ? x : E.A; // error 
}
```
Expected: no errors
Actual: enum isn't a subtype or assignable to Object or {}

**Comment by jbondc** [comment]
I'm confused what should be the behavior?
```

    interface I19 {
        [x: string]: Object;
        foo: E; // BUG 831833
        ~~~~~~~
!!! error TS2411: Property 'foo' of type 'E' is not assignable to string index type 'Object'.
    }
    
    
    interface I20 {
        [x: string]: {};
        foo: E; // BUG 831833
        ~~~~~~~
!!! error TS2411: Property 'foo' of type 'E' is not assignable to string index type '{}'.
    }
```
Is that correct?

**Comment by RyanCavanaugh** [comment]
I believe that bug (and thus that test) refer to out-of-date behavior. All types are assignable to `{}` and `Object`, so those shouldn't be an error.
